### PR TITLE
IPB-7683 - extract timings from har file§

### PR DESCRIPTION
**Pull request by Sarah-J-S**
Changes added to extract timings from har file.

NB the sleep time for the first entry will always be zero as there is no preceding time. I have assumed that non-zero user supplied sleep times take precedence over captured times.

### CODE REVIEW

**Comment by jb098** [comment]
Intial code looks good. I think it would be worth adding a unit test for this before we merge anything.

**Comment by Sarah-J-S** [comment]
Indeed it would.

**Comment by Sarah-J-S** [comment]
Well, it's a test. I'm not proud of it but it does what I think it needs to do. Please review with extreme prejudice.

File: test/test_har_to_mite.py [hunk 1]
```diff
@@ -182,42 +183,58 @@ def test_set_headers():
...
+    if os.path.exists(testfile):
+        os.remove(testfile)
+    if os.path.exists(testfile10secs):
+        os.remove(testfile10secs)
+
+    har_convert_to_mite("test.har", "testharconvmite.txt", 0)
+    har_convert_to_mite("test.har", "testharconvmite10secs.txt", 10)
+
+    assert os.path.exists(testfile)
+    assert os.path.exists(testfile10secs)
```
Comment (aecay) [inline test/test_har_to_mite.py:223]: I think we want some assertions on the new behavior that you have introduced -- i.e. do the delays in the generated code match those in the har file?

Comment (Sarah-J-S) [reply test/test_har_to_mite.py:223]: I agree. While I couldn't see a way to do that without reams of code on Friday, I may have a quicker and dirtier method in mind after a weekend's reflection.

**Comment by Sarah-J-S** [comment]
Extra testing added to ensure that the extracted wait times are what we expect to see.

File: test/test_har_to_mite.py (hunk 1 above)
Comment (Sarah-J-S) [reply test/test_har_to_mite.py:223]: Quick and dirty test now in place.

File: test/test_har_to_mite.py [hunk 2]
```diff
@@ -182,42 +183,64 @@ def test_set_headers():
...
+
+    with open(testfile, "r") as testharoutfile:
+        testharStr = testharoutfile.read()
+    assert testharStr.find("await sleep(0)") != -1
```
Comment (aecay) [inline test/test_har_to_mite.py:227]:
```suggestion
    assert "await sleep(0)" in testharStr
```

**Review by aecay — APPROVED** [review]
One suggested tweak to the test assertions but this lgtm

File: test/test_har_to_mite.py (hunk 1 above)
Comment (aecay) [reply test/test_har_to_mite.py:223]: Brilliant, thanks 🙂

File: test/test_har_to_mite.py (hunk 2 above)
Comment (aecay) [reply test/test_har_to_mite.py:227]: (and so on for the other assertions)

Comment (Sarah-J-S) [reply test/test_har_to_mite.py:227]: That looks neater. I shall make it so. Another thing learned about Python.

File: test/test_har_to_mite.py [hunk 3]
```diff
@@ -182,42 +183,64 @@ def test_set_headers():
...
+        os.remove(testfile10secs)
+
+    har_convert_to_mite("test.har", "testharconvmite.txt", 0)
+    har_convert_to_mite("test.har", "testharconvmite10secs.txt", 10)
+
+    assert os.path.exists(testfile)
+    assert os.path.exists(testfile10secs)
+
+    with open(testfile, "r") as testharoutfile:
+        testharStr = testharoutfile.read()
```
Comment (jb098) [inline test/test_har_to_mite.py:226]: One very small comment that it's more pythonnic to use snake_case but I'll send you the tick anyway.

**Review by jb098 — APPROVED** [review]

File: test/test_har_to_mite.py (hunk 3 above)
Comment (Sarah-J-S) [reply test/test_har_to_mite.py:226]: My future code will strive for greater pythonicity. At least I've given up Hungarian notation.

**Pull request merged**